src/Common/src/System/Net/NetworkInformation/UnixCommandLinePing.cs

```diff
@@ -17,6 +18,7 @@ internal static class UnixCommandLinePing
 
         private static readonly string s_discoveredPing4UtilityPath = GetPingUtilityPath(ipv4: true);
         private static readonly string s_discoveredPing6UtilityPath = GetPingUtilityPath(ipv4: false);
+        private static readonly bool s_isOSX = Environment.OSVersion.Platform == PlatformID.MacOSX;
```

Comment:
Did you test this and it worked?  I thought we returned PlatformID.Unix for macOS.